pkg/sql/opt/optbuilder/project.go

```diff
@@ -224,6 +225,22 @@ func (b *Builder) finishBuildScalar(
 		return scalar
 	}
 
+	// Add a view dependency on a RegClass object.
+	if b.trackViewDeps {
+		if texpr.ResolvedType() == types.RegClass {
+			regclass, err := texpr.Eval(b.evalCtx)
```

Comment:
You can only do this if `texpr` doesn't contain variables right? Wouldn't this query error out?

```
CREATE VIEW vregclass AS SELECT x::regclass FROM (SELECT 'toreg' AS x)
```

Reply:
Yeah you're right, need to figure out how to handle this case.

Reply:
Interesting case here where Postgres doesn't actually add a dependency on table t when creating a view in the way you mentioned.

d=# create view vreg as select x::regclass from (select 't' as x) as f;
CREATE VIEW
d=# select * from vreg;
 x 
---
 t
(1 row)

d=# drop table t;
DROP TABLE
d=#

Reply:
Changed it to not add dependencies when the regclass expression contains variables, this seems to match what pg does.